src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Snippets/XmlSnippetParser.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.VisualStudio.Editor.Razor.Snippets;
+
+internal static partial class XmlSnippetParser
+{
+
+    internal static ParsedXmlSnippet? GetParsedXmlSnippet(SnippetInfo matchingSnippetInfo, ILogger? logger = null)
+    {
+        ParsedXmlSnippet? parsedSnippet = null;
+        try
+        {
+            logger?.LogInformation($"Reading snippet for {matchingSnippetInfo.Title} with path {matchingSnippetInfo.Path}");
+            parsedSnippet = GetAndParseSnippetFromFile(matchingSnippetInfo);
+        }
+        catch (Exception ex)
+        {
+            logger?.LogCritical(ex, $"Got exception parsing xml snippet {matchingSnippetInfo.Title} from file {matchingSnippetInfo.Path}");
+        }
+
+        return parsedSnippet;
+    }
+
+    private static ParsedXmlSnippet GetAndParseSnippetFromFile(SnippetInfo snippetInfo)
+    {
+        // Read the XML file to get the snippet and snippet metadata.
+        var matchingSnippet = RetrieveSnippetXmlFromFile(snippetInfo);
+
+        Debug.Assert(matchingSnippet.IsExpansionSnippet(), "Only expansion snippets are supported");
```

Comment:
Consider removing the `Debug.Assert` and putting the message in the `InvalidOperationException` below.